src/binary_heap.rs

```diff
@@ -389,7 +389,25 @@ where
 
     /// Removes the *top* (greatest if max-heap, smallest if min-heap) item from the binary heap and
     /// returns it, without checking if the binary heap is empty.
-    #[allow(clippy::missing_safety_doc)] // TODO
+    ///
+    /// # Safety
+    ///
+    /// The binary heap must not be empty.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use heapless::binary_heap::{BinaryHeap, Max};
+    ///
+    /// let mut heap: BinaryHeap<_, Max, 8> = BinaryHeap::new();
+    /// heap.push(42).unwrap();
+    ///
+    /// if !heap.is_empty() {
+    ///     // SAFETY: the heap is not empty
```

Comment:
I'd skip the `is_empty` check here since you wouldn't use it in a real use case for this, and argue with

```suggestion
    ///     // SAFETY: We just pushed a number onto the heap, so it cannot be empty.
```

Reply:
Good point, fixed!